### ISSUE TITLE
SW does not even attempt to register

### ISSUE THREAD

**Issue by albgen** [vite-pwa/vite-plugin-pwa#591]
Hi,

Have a project with vite and installed vite-plugin-pwa. Long story short, simply workbox does not work. The service worker does not even try to register.  I don't see errors or else, i don't see even the attempt. Tried i don't know how many configs but still nothing. The generated build is correct.

Then i thought, ok let's download the samples here and run one of them. Surprise! Nope, even that one is working.
I tried react-router sample. 
Now, since i'm on windows, i just added a script "win-build": "vite build" and one simpe "win-start": "vite"

It compiled and produced the files:
![image](https://github.com/vite-pwa/vite-plugin-pwa/assets/7231968/0c94edbb-278a-44d0-acd4-3934a45c34ab)

than i run "vite", i see the sample website but opening with chrome devtools, on the application tab i don't see nothing registered as service worker.

This is the same behaviour as my project. Tried also on Edge browser but same identical behaviour.

What is wrong? Any idea?

**Comment by userquin** [comment]
I use Windows 10 PRO and all examples in the repo working (also the tests).

Provide a minimal repro or at least pwa configuration...

**Comment by albgen** [comment]
exactly this one https://github.com/vite-pwa/vite-plugin-pwa/tree/main/examples/react-router
The only modification is that i added on the scripts part of the package.json the following

"win-build": "vite build",
"win-start": "vite"

then:
 npm start win-build
 npm start win-start

the sample is started, working normally but not the registration of the service worker

**Comment by userquin** [comment]
you need to run the preview script, running build will build the app but running `vite` will start the dev server, add this script and run it: 
`"win-start-preview": "vite build && vite preview"

**Comment by albgen** [comment]
yes it works with the preview paramter, but why without does not?

**Comment by userquin** [comment]
dev options only to test web manifest and custom sw logic (injectManifest strategy), you should disable dev options once tested web manifest

**Comment by albgen** [comment]
HMR does not work with "vite preview".
I can't make changes and test offline capabilites.

Any workaround for this?

**Comment by userquin** [comment]
I think you are mixing concepts, you don't need HMR (which you only have it in development) to test your application offline, just run the build + preview script and once the service worker is installed and activated, put the browser offline in the network tab.

You should only use the PWA plugin in development to test that the web manifest is ok (you're not using custom sw), once tested you should disable it. For this reason (HMR), the PWA plugin does not include anything other than base path in the sw precache manifest.

If you want to check how the application is updated, you can try changing some text in the index.html for example (change the title) and run the build + preview script and once preview server started refresh the page, remember that something must change between builds, otherwise the application will remain the same (you can open Application > Service workers, check the `#id`, should change => new sw will appear in yellow, if you're using auto update, it will be installed and activated, otherwise will be in skip waiting state until update button clicked).

**Comment by userquin** [comment]
https://vite-pwa-org.netlify.app/guide/periodic-sw-updates.html

**Comment by albgen** [comment]
> https://vite-pwa-org.netlify.app/guide/periodic-sw-updates.html

seems this is needed to update every hour and in specific cases. My problem is when developing & HMR.

**Comment by userquin** [comment]
when developing, Vite will update the app via HMR
